app/uk/gov/hmrc/customs/emailfrontend/services/EmailCacheService.scala

```diff
@@ -52,50 +53,34 @@ class EmailCacheService @Inject()(caching: Save4LaterCachingConfig, applicationC
-
-  def saveEmail(internalId: InternalId, emailStatus: EmailStatus)
-               (implicit hc: HeaderCarrier, executionContext: ExecutionContext): Future[CacheMap] = {
-    Logger.info("saving email address to save 4 later")
-    cache[EmailStatus](internalId.id, emailKey, emailStatus)
-  }
-
-  def saveTimeStamp(internalId: InternalId, verifiedEmailTimestamp: DateTime)
-                   (implicit hc: HeaderCarrier, executionContext: ExecutionContext): Future[CacheMap] = {
-
-    import uk.gov.hmrc.customs.emailfrontend.DateTimeUtil._
-    Logger.info("saving verified email time stamp to save 4 later")
-    cache[DateTime](internalId.id, timestampKey, verifiedEmailTimestamp)
-  }
-
-  def fetchTimeStamp(internalId: InternalId)(implicit hc: HeaderCarrier, executionContext: ExecutionContext): Future[Option[DateTime]] = {
-    import uk.gov.hmrc.customs.emailfrontend.DateTimeUtil._
-    Logger.info("retrieving cached timestamp from save 4 later")
-    fetchAndGetEntry[DateTime](internalId.id, timestampKey)
-  }
-
   def remove(internalId: InternalId)
             (implicit hc: HeaderCarrier, executionContext: ExecutionContext): Future[HttpResponse] = {
     Logger.info("removing cached data from save 4 later")
     remove(internalId.id)
   }
 
-  def fetchEmail(internalId: InternalId)(implicit hc: HeaderCarrier, executionContext: ExecutionContext): Future[Option[EmailStatus]] = {
-    Logger.info("retrieving cached email from save 4 later")
-    fetchAndGetEntry[EmailStatus](internalId.id, emailKey)
+  def save(internalId: InternalId, cachedData: CachedData)
+          (implicit hc: HeaderCarrier, executionContext: ExecutionContext): Future[CacheMap] = {
+    Logger.info("saving email address and timestamp to save 4 later")
+    cache[CachedData](internalId.id, emailKey, cachedData)
+  }
+
+  def fetch(internalId: InternalId)(implicit hc: HeaderCarrier, executionContext: ExecutionContext): Future[Option[CachedData]] = {
+    Logger.info("retrieving email address and timestamp from save 4 later")
+    fetchAndGetEntry[CachedData](internalId.id, emailKey)
   }
 }
 
+object EmailCacheService {
+
+  implicit class EmailCacheServiceHelper(emailCacheService: EmailCacheService) {
 
-object EmailCacheService{
-  implicit class EmailCacheServiceHelper(emailCacheService: EmailCacheService){
-    def emailAmendmentStatus(internalId: InternalId)(implicit hc: HeaderCarrier, executionContext: ExecutionContext): Future[EmailAmendmentStatus] = {
-      emailCacheService.fetchTimeStamp(internalId).flatMap {
-        case Some(date) => if (date.isBefore(DateTime.now.minusDays(1))) {
-          emailCacheService.remove(internalId).map(_ => AmendmentCompleted )
-        } else Future.successful(AmendmentInProgress)
-        case None => Future.successful(AmendmentNotDetermined)
+    def emailAmendmentData(internalId: InternalId)(redirectBasedOnEmailStatus: String => Future[Result])(implicit hc: HeaderCarrier, executionContext: ExecutionContext) = {
```

Comment:
This can be avoided, I think I discussed with you this that we should have isAmended function on  EmailDetails currently called as CachedData in this PR I would prefer  EmailDetails or EmailStatus

Reply:
yes, I did add that function/val in CachedData as `amendmentInProgress`